File: backend/src/main/scala/domains/auth/model/UpdateManagedUserSettingsRequest.scala
package domains.auth.model

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UpdateManagedUserSettingsRequest(
  displayName: DisplayName,
  email: EmailAddress,
  preferences: UserPreferences,
  newPassword: Option[PlaintextPassword]
)

object UpdateManagedUserSettingsRequest:
  given Encoder[UpdateManagedUserSettingsRequest] = deriveEncoder[UpdateManagedUserSettingsRequest]
  given Decoder[UpdateManagedUserSettingsRequest] = deriveDecoder[UpdateManagedUserSettingsRequest]

File: frontend/src/features/auth/model/UpdateManagedUserSettingsRequest.ts
import type { DisplayName, EmailAddress, PlaintextPassword } from '@/features/auth/model/AuthValues'
import type { UserPreferences } from '@/features/auth/model/UserPreferences'

export type UpdateManagedUserSettingsRequest = {
  displayName: DisplayName
  email: EmailAddress
  preferences: UserPreferences
  newPassword: PlaintextPassword | null
}
